addons/purchase/controllers/portal.py

```diff
@@ -35,8 +40,66 @@ def resize_to_48(b64source):
         values = {
             'order': order,
             'resize_to_48': resize_to_48,
+            'report_type': 'html',
+        }
+        if order.state in ('sent'):
+            history = 'my_rfqs_history'
+        else:
+            history = 'my_purchases_history'
+        return self._get_page_view_values(order, access_token, values, history, False, **kwargs)
+
+    @http.route(['/my/rfq', '/my/rfq/page/<int:page>'], type='http', auth="user", website=True)
+    def portal_my_requests_for_quotation(self, page=1, date_begin=None, date_end=None, sortby=None, filterby=None, **kw):
```

Comment:
This method is almost the same than `portal_my_purchase_orders`.
You could split them into two methods (a first one to get the domain and a second one to get the values) and call them in `portal_my_requests_for_quotation` and `portal_my_purchase_orders` while still doing the specific RFQ/PO stuff in each route methods.